### DISCUSSION TITLE
Ignore parts of the line?

### DISCUSSION THREAD

**Discussion by Pierre-de-la-Verre** [WinMerge/winmerge#550] [Q&A] [answered]
If I understand it correctly the "line filter" handles the entire line and not the part of it.

Example
File a:     "line=1234, message="This is my message"
File b:     "line=1456, message="This is my message"

If I compare it with standard settings, it would show a difference because "line=.." is different

When I activate a "line filter" with definition "line=[0-9]*," then there will be no difference.

But I have these data:
File a:     "line=1234, message="This is my message"
File b:     "line=1456, message="This is another message"

What to do now?
I want to skip the difference in the "line=... section, but I need the difference in the "message=.. section"

Is there a "ignore part of the line"-feature?
Do I have to edit it before?
What else?

Thanks and regards!

**Comment by sdottaka** [comment, marked as answer]
We've recently added the features you're looking for, but there are a few things we'd like to fix, so I'm not sure if we'll be in time for the new version we're releasing this month.

Until the feature is released, please use the PrediffLineFilter.sct plugin installed in the current WinMerge as follows.

![551](https://user-images.githubusercontent.com/98126/104387312-29c3c900-557a-11eb-976e-3417bd66ef33.gif)

**Comment by Pierre-de-la-Verre** [comment]
Thanks for your reply and the image
looks fine. First testing is OK!

**Comment by Pierre-de-la-Verre** [comment]
Works fine, thanks.

Is it also possible to activate the plugin permanently?

**Comment by sdottaka** [comment]
To apply a Prediffer plugin like PrediffLineFilter.sct automatically, select the Plugin-> Automatic Prediffer menu item.

![image](https://user-images.githubusercontent.com/98126/105565408-f9dea780-5d69-11eb-8113-991cddabd3ea.png)

However, the PrediffLineFilter.sct plugin is automatically applied only to files with a .nomatch extension by default.
To change this, in the Plugins dialog, specify the pattern as a regular expression as shown below.

The following is an example that matches the extensions .txt and .HTML.

![image](https://user-images.githubusercontent.com/98126/105565509-838e7500-5d6a-11eb-893e-283cf09e66ec.png)
